openmetadata-ui/src/main/resources/ui/src/components/Alerts/AlertsDetails/AlertDetails.test.tsx

```diff
@@ -0,0 +1,139 @@
+  jest.fn().mockReturnValue(<span>TitleBreadcrumb</span>)
+);
+
+jest.mock('../../PageHeader/PageHeader.component', () =>
+  jest.fn().mockReturnValue(<span>PageHeader</span>)
+);
+
+const mockDelete = jest.fn();
+const mockPush = jest.fn();
+
+const mockProps1 = {
+  alerts: {
+    id: 'alertId',
+    name: 'alertName',
+    alertType: AlertType.ActivityFeed,
+    destinations: [] as Destination[],
+    filteringRules: {
+      resources: ['resource1', 'resource2', 'resource3'],
+      rules: [
+        {
+          fullyQualifiedName: 'conditionName',
+          condition: 'condition',
+          effect: Effect.Include,
+          name: 'ruleName',
+        },
+      ],
+    },
+  },
+  onDelete: mockDelete,
+};
+
+const mockProps2 = {
+  ...mockProps1,
+  pageHeaderData: {} as HeaderProps['data'],
+  allowDelete: false,
+  breadcrumb: {} as TitleBreadcrumbProps['titleLinks'],
+  allowEdit: false,
+};
+
+describe('AlertDetailsComponent', () => {
+  it('should render all necessary elements based on mockProps1', () => {
+    render(<AlertDetailsComponent {...mockProps1} />, {
+      wrapper: MemoryRouter,
+    });
+
+    expect(screen.queryByText('TitleBreadcrumb')).not.toBeInTheDocument();
+    expect(screen.queryByText('PageHeader')).not.toBeInTheDocument();
+
+    const editButton = screen.getByRole('button', { name: 'label.edit' });
+    const deleteButton = screen.getByRole('button', { name: 'label.delete' });
+
+    expect(editButton).toBeInTheDocument();
+    expect(deleteButton).toBeInTheDocument();
+
+    expect(screen.getByText('label.trigger')).toBeInTheDocument();
+    expect(
+      screen.getByText('resource1, resource2, resource3')
+    ).toBeInTheDocument();
+
+    expect(screen.getByText('label.filter-plural')).toBeInTheDocument();
+
+    expect(screen.getByText('conditionName === condition')).toBeInTheDocument();
+
+    expect(screen.getByText('label.destination')).toBeInTheDocument();
+  });
+
+  it('should render breadcrumb and pageHeader if data passed in props', () => {
+    render(<AlertDetailsComponent {...mockProps2} />, {
+      wrapper: MemoryRouter,
+    });
+
+    expect(screen.getByText('TitleBreadcrumb')).toBeInTheDocument();
+    expect(screen.getByText('PageHeader')).toBeInTheDocument();
+
+    const editButton = screen.queryByRole('button', { name: 'label.edit' });
+    const deleteButton = screen.queryByRole('button', { name: 'label.delete' });
+
+    expect(editButton).not.toBeInTheDocument();
+    expect(deleteButton).not.toBeInTheDocument();
+  });
+
+  it('should call action handlers on click of buttons', () => {
+    const history = createMemoryHistory();
+
+    history.push = mockPush;
+
+    render(
+      <Router history={history}>
+        <AlertDetailsComponent {...mockProps1} />
+      </Router>
+    );
+
+    const editButton = screen.getByRole('button', { name: 'label.edit' });
+    const deleteButton = screen.getByRole('button', { name: 'label.delete' });
+
+    userEvent.click(deleteButton);
```

Comment:
Can we wrap this within `act`? please take care of this in all the places.

Reply:
can you eleborate more why we want to wrap the userEvent.click in act at every place. It should be wrapped in act only if click event is doing state update.

Reply:
Yes, we should wrap the event only if we are updating the state.

By this, I meant we should check all the places and if we are updating the state on some event then it should be wrapped in `act`.
> Please take care of this in all the places.